### PR TITLE
Rename wg-rls-2 to rust-analyzer and make it a team

### PR DESCRIPTION
**Pull request by Veykril**
Renames the wg-rls-2 and wg-rls-2-triage teams to wg-rust-analyzer and wg-rust-analyzer-contributors. Also gives the wg-rust-analyzer-contributors team bors permissions, as the core team is running out of reviewing hands.

### CODE REVIEW

File: teams/wg-rust-analyzer-contributors.toml
Comment (rylev) [inline teams/wg-rust-analyzer-contributors.toml:1]: Would it make sense to just call the team rust-analyzer (and make it a team instead of a working group)? cc @pnkfelix @wesleywiser

File: teams/rust-analyzer.toml
Comment (rylev) [inline teams/rust-analyzer.toml:1]: We should probably rename this as well, no?

Comment (Veykril) [reply teams/rust-analyzer.toml:1]: Oh I missed that

**Comment by rylev** [comment]
CI is failing now because there are some members of the rust-analyzer team who don't have email addresses but with r-a becoming a team they now join the all@ mailing list. 

I'd like to get approval of the change of r-a from a working group to a subteam from @pnkfelix and/or @wesleywiser before we merge.

**Comment by Veykril** [comment]
As rylev pointed out, turning the r-a working groups into teams requires all active members to supply an e-mail:
Mails are missing for @edwin0cheng, @lowr and @lnicola. Are you fine with adding mails to your people files (or handing them to me so I can add them in this PR)?

**Comment by Veykril** [comment]
On another note, do we support subteams of subteams? (i.e. making the contributors team a subteam of the r-a team which is a subteam of compiler?). Didn't make that change here because I assume that doesn't work and it not really mattering.

**Comment by rylev** [comment]
> On another note, do we support subteams of subteams? (i.e. making the contributors team a subteam of the r-a team which is a subteam of compiler?). Didn't make that change here because I assume that doesn't work and it not really mattering.

We prevent subteams of subteams though I'm not 100% sure why. Let's keep it as a subteam of compiler and we can think about improving this later.

**Comment by lowr** [comment]
Filed #943, but you can cherry-pick and close it if that's faster.

**Comment by Veykril** [comment]
Probably easier if I pick it over 👍

File: teams/rust-analyzer.toml
Comment (wesleywiser) [inline teams/rust-analyzer.toml:11]: Nit: maybe update the description too since we're renaming it?

**Review by wesleywiser — APPROVED** [review]
@pnkfelix and I discussed this and we're both 👍 on the idea!

File: teams/rust-analyzer.toml
Comment (rylev) [reply teams/rust-analyzer.toml:11]: @Veykril Does this description still fit or shall we update it?

Comment (lnicola) [reply teams/rust-analyzer.toml:11]: Maybe "Rust IDE support and error-resilient compiler front-end" or something like? @matklad?

I think Veykril is less around these days, not sure if I can push to this branch.

Comment (Veykril) [reply teams/rust-analyzer.toml]: Updated it to lnicola's suggestion

**Review by lnicola — APPROVED** [review]

**Review by rylev — APPROVED** [review]

**Pull request merged**
